### PR TITLE
Pybind11 update to 2.4.2

### PR DESCRIPTION
**Pull request by breznak**
from 2.2.4 to 2.3.0, about 2 years of updates. New release this summer.

### CODE REVIEW

File: external/pybind11.cmake [hunk 1]
```diff
@@ -20,7 +20,7 @@
 if(EXISTS "${REPOSITORY_DIR}/build/ThirdParty/share/pybind11.tar.gz")
     set(URL "${REPOSITORY_DIR}/build/ThirdParty/share/pybind11.tar.gz")
 else()
-    set(URL https://github.com/pybind/pybind11/archive/v2.2.4.tar.gz)
+    set(URL https://github.com/pybind/pybind11/archive/v2.3.0.tar.gz)
```
Comment (dkeeney) [inline external/pybind11.cmake]: Oh, very good.
Thanks.

**Comment by breznak** [comment]
hmm, I think the OSX compilation (due to a warning) err is fixed in pybind master. I'll see if it's reasonable to silence the warning there, or just wait for 2.3.1 ?

File: .github/workflows/htmcore.yml [hunk 2]
```diff
@@ -92,6 +92,7 @@ jobs:
         path: build/Release/distr/dist
 
     - name: Test python package is working
+      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
```
Comment (breznak) [inline .github/workflows/htmcore.yml]: fixing an unrelated CI failuer introduced in other PR, this breaks all current PRs

**Comment by breznak** [comment]
FYI @dkeeney merged to fix issue with PRs failing in CI (missing /distr/dist/ in `ls`)

**Pull request merged**